catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -557,23 +556,18 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
       // then based on
       // those names we can obtain metadata for each individual table and get the type we needed.
       List<String> allTables = clientPool.run(c -> c.getAllTables(schemaIdent.name()));
-      return clientPool.run(
-          c ->
-              c.getTableObjectsByName(schemaIdent.name(), allTables).stream()
-                  .filter(
-                      tb -> {
-                        boolean isSupportTable = SUPPORT_TABLE_TYPES.contains(tb.getTableType());
-                        if (!isSupportTable) {
-                          return false;
-                        }
-                        if (!listAllTables) {
-                          Map<String, String> parameters = tb.getParameters();
-                          return isHiveTable(parameters);
-                        }
-                        return true;
-                      })
-                  .map(tb -> NameIdentifier.of(namespace, tb.getTableName()))
-                  .toArray(NameIdentifier[]::new));
+      if (!listAllTables) {
+        String filter =
+            String.format(
+                "%stable_type = \"ICEBERG\"", hive_metastoreConstants.HIVE_FILTER_FIELD_PARAMS);
```

Comment:
Filtering out only the Iceberg table seems to be insufficient, and we currently also do not support external and view types.
So I think you should use show managed type table when `listAllTables=false`

Reply:
I found the listTableNamesByFilter can’t support filter by table type.so , this method can't meet the requirement of filting view tables.
<img width="824" alt="image" src="https://github.com/user-attachments/assets/4b3a02d0-394b-4b11-a79c-09d58511a635">

Reply:
If we do not filter the view, what impact will it have on Gravitino?

Reply:
> If we do not filter the view, what impact will it have on Gravitino?

I have tested it.The view table ,don‘t have any impact ,so i think we can do like this.

Reply:
I add some comment ,please review again

Reply:
Now,the ci issue is a troublesome problem. https://github.com/apache/gravitino/actions/runs/10452698362/job/28942732918?pr=4469
 This ci issue is due to a bug in hive, and the iceberg project has the same proble.
https://github.com/apache/iceberg/pull/2722#issuecomment-867363019
<img width="1240" alt="image" src="https://github.com/user-attachments/assets/4b49b466-dc17-4749-89c4-1f7267aa90e7">

This hive bug is only happen in Derby. In our environment , the metastore storage is mysql,so we don't encounter this problem.